src/mono/mono/mini/debugger-agent.c

```diff
@@ -262,231 +262,18 @@ typedef struct {
-	CMD_EVENT_REQUEST_SET = 1,
-	CMD_EVENT_REQUEST_CLEAR = 2,
-	CMD_EVENT_REQUEST_CLEAR_ALL_BREAKPOINTS = 3
-} CmdEvent;
-
-typedef enum {
-	CMD_COMPOSITE = 100
-} CmdComposite;
-
-typedef enum {
-	CMD_APPDOMAIN_GET_ROOT_DOMAIN = 1,
-	CMD_APPDOMAIN_GET_FRIENDLY_NAME = 2,
-	CMD_APPDOMAIN_GET_ASSEMBLIES = 3,
-	CMD_APPDOMAIN_GET_ENTRY_ASSEMBLY = 4,
-	CMD_APPDOMAIN_CREATE_STRING = 5,
-	CMD_APPDOMAIN_GET_CORLIB = 6,
-	CMD_APPDOMAIN_CREATE_BOXED_VALUE = 7,
-	CMD_APPDOMAIN_CREATE_BYTE_ARRAY = 8,
-} CmdAppDomain;
-
-typedef enum {
-	CMD_ASSEMBLY_GET_LOCATION = 1,
-	CMD_ASSEMBLY_GET_ENTRY_POINT = 2,
-	CMD_ASSEMBLY_GET_MANIFEST_MODULE = 3,
-	CMD_ASSEMBLY_GET_OBJECT = 4,
-	CMD_ASSEMBLY_GET_TYPE = 5,
-	CMD_ASSEMBLY_GET_NAME = 6,
-	CMD_ASSEMBLY_GET_DOMAIN = 7,
-	CMD_ASSEMBLY_GET_METADATA_BLOB = 8,
-	CMD_ASSEMBLY_GET_IS_DYNAMIC = 9,
-	CMD_ASSEMBLY_GET_PDB_BLOB = 10,
-	CMD_ASSEMBLY_GET_TYPE_FROM_TOKEN = 11,
-	CMD_ASSEMBLY_GET_METHOD_FROM_TOKEN = 12,
-	CMD_ASSEMBLY_HAS_DEBUG_INFO = 13,
-	CMD_ASSEMBLY_GET_CATTRS = 14,
-} CmdAssembly;
-
-typedef enum {
-	CMD_MODULE_GET_INFO = 1,
-} CmdModule;
-
-typedef enum {
-	CMD_FIELD_GET_INFO = 1,
-} CmdField;
-
-typedef enum {
-	CMD_METHOD_GET_NAME = 1,
-	CMD_METHOD_GET_DECLARING_TYPE = 2,
-	CMD_METHOD_GET_DEBUG_INFO = 3,
-	CMD_METHOD_GET_PARAM_INFO = 4,
-	CMD_METHOD_GET_LOCALS_INFO = 5,
-	CMD_METHOD_GET_INFO = 6,
-	CMD_METHOD_GET_BODY = 7,
-	CMD_METHOD_RESOLVE_TOKEN = 8,
-	CMD_METHOD_GET_CATTRS = 9,
-	CMD_METHOD_MAKE_GENERIC_METHOD = 10
-} CmdMethod;
-
-typedef enum {
-	CMD_TYPE_GET_INFO = 1,
-	CMD_TYPE_GET_METHODS = 2,
-	CMD_TYPE_GET_FIELDS = 3,
-	CMD_TYPE_GET_VALUES = 4,
-	CMD_TYPE_GET_OBJECT = 5,
-	CMD_TYPE_GET_SOURCE_FILES = 6,
-	CMD_TYPE_SET_VALUES = 7,
-	CMD_TYPE_IS_ASSIGNABLE_FROM = 8,
-	CMD_TYPE_GET_PROPERTIES = 9,
-	CMD_TYPE_GET_CATTRS = 10,
-	CMD_TYPE_GET_FIELD_CATTRS = 11,
-	CMD_TYPE_GET_PROPERTY_CATTRS = 12,
-	CMD_TYPE_GET_SOURCE_FILES_2 = 13,
-	CMD_TYPE_GET_VALUES_2 = 14,
-	CMD_TYPE_GET_METHODS_BY_NAME_FLAGS = 15,
-	CMD_TYPE_GET_INTERFACES = 16,
-	CMD_TYPE_GET_INTERFACE_MAP = 17,
-	CMD_TYPE_IS_INITIALIZED = 18,
-	CMD_TYPE_CREATE_INSTANCE = 19,
-	CMD_TYPE_GET_VALUE_SIZE = 20
-} CmdType;
-
-typedef enum {
-	CMD_STACK_FRAME_GET_VALUES = 1,
-	CMD_STACK_FRAME_GET_THIS = 2,
-	CMD_STACK_FRAME_SET_VALUES = 3,
-	CMD_STACK_FRAME_GET_DOMAIN = 4,
-	CMD_STACK_FRAME_SET_THIS = 5,
-} CmdStackFrame;
-
-typedef enum {
-	CMD_ARRAY_REF_GET_LENGTH = 1,
-	CMD_ARRAY_REF_GET_VALUES = 2,
-	CMD_ARRAY_REF_SET_VALUES = 3,
-} CmdArray;
-
-typedef enum {
-	CMD_STRING_REF_GET_VALUE = 1,
-	CMD_STRING_REF_GET_LENGTH = 2,
-	CMD_STRING_REF_GET_CHARS = 3
-} CmdString;
-
-typedef enum {
-	CMD_POINTER_GET_VALUE = 1
-} CmdPointer;
-
-typedef enum {
-	CMD_OBJECT_REF_GET_TYPE = 1,
-	CMD_OBJECT_REF_GET_VALUES = 2,
-	CMD_OBJECT_REF_IS_COLLECTED = 3,
-	CMD_OBJECT_REF_GET_ADDRESS = 4,
-	CMD_OBJECT_REF_GET_DOMAIN = 5,
-	CMD_OBJECT_REF_SET_VALUES = 6,
-	CMD_OBJECT_REF_GET_INFO = 7,
-} CmdObject;
+#define MAJOR_VERSION 3
+#define MINOR_VERSION 0
```

Comment:
nit: the protocol didn't change in this PR, right? it's changing in the next PR.  So the version bump is not needed yet.

Should we have the protocol version in `debugger-protocol.h`, too?

Reply:
Yes, I moved to debugger-protocol.h